jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1537,6 +1539,9 @@ public void failed(Throwable failure)
             try (AutoLock ignored = _request._lock.lock())
             {
                 httpChannelState = _request._httpChannelState;
+                if (httpChannelState == null)
```

Comment:
I think you could just call this and it will do the null check.
```java
httpChannelState = _request.lockedGetHttpChannelState();
```

Reply:
`_request.lockedGetHttpChannelState()` throws if the httpChannelState is null. We don't want to do that in a callback's `failed()` method.

Reply:
@lorban are you sure?   If we want multiple calls to a callback to be a noop, then the throw in _request.lockedGetHttpChannelState() is wrong.     Since that is a throw, doesn't it indicate that we are throwing elsewhere?

We should decide one way or another and javadoc in the Callback class if we should throw or noop.

Reply:
I think @lachlan-roberts meant `lockedCompleteCallback()` which does the null check, and is already called a few lines afterwards. Moving the call before the `_request._httpChannelState` deref should be all that's needed.